backend/src/core/middlewares/auth.middleware.ts

```diff
@@ -110,48 +115,70 @@ export const InitAuthMiddleware = (authService: AuthService) => {
     return res.json({})
   }
 
-  /**
-   * Checks that request has a valid cookie (based on session), or a valid Authorization header
-   * @param req
-   * @param res
-   * @param next
-   */
-  const isCookieOrApiKeyAuthenticated = async (
-    req: Request,
-    res: Response,
-    next: NextFunction
-  ): Promise<Response | void> => {
-    try {
-      if (authService.checkCookie(req)) {
-        logger.info({
-          message: 'User authenticated by cookie',
-          action: 'isCookieOrApiKeyAuthenticated',
-          user: req.session?.user,
-        })
-        return next()
+  type AuthenticatorType = (req: Request) => Promise<boolean>
+  const authenticators: Record<AuthType, AuthenticatorType> = {
```

Comment:
wow this is some insane use of generics to do what i would've written as a basic switch case to handle. nice.